src/builtins/BuiltinIterator.cpp

```diff
@@ -430,6 +485,91 @@ static Value builtinIteratorSome(ExecutionState& state, Value thisValue, size_t
+        return d;
+    }
+    return (d < 0 ? -1 : 1) * std::floor(std::abs(d));
+}
+
+static std::pair<Value, bool> iteratorTakeClosure(ExecutionState& state, IteratorHelperObject* obj, void* data)
+{
+    // Let closure be a new Abstract Closure with no parameters that captures iterated and integerLimit and performs the following steps when called:
+    // Let remaining be integerLimit.
+    // Repeat,
+    //     If remaining = 0, then
+    //        Return ? IteratorClose(iterated, ReturnCompletion(undefined)).
+    //     If remaining ≠ +∞, then
+    //         Set remaining to remaining - 1.
+    //     Let value be ? IteratorStepValue(iterated).
+    //     If value is done, return ReturnCompletion(undefined).
+    //     Let completion be Completion(Yield(value)).
+    //     IfAbruptCloseIterator(completion, iterated).
+    IteratorData* closureData = reinterpret_cast<IteratorData*>(data);
+    IteratorRecord* iterated = obj->underlyingIterator();
+    double remaining = closureData->callback.toNumber(state);
+
+    if (remaining == 0) {
+        if (!iterated->m_done) {
+            iterated->m_done = true;
+            IteratorObject::iteratorClose(state, iterated, Value(), false);
+        }
+        return std::make_pair(Value(), true);
+    }
+    if (remaining != std::numeric_limits<double>::infinity()) {
+        closureData->callback = Value(static_cast<int32_t>(remaining - 1));
+    }
+
+    auto value = IteratorObject::iteratorStepValue(state, iterated);
+    if (!value) {
+        iterated->m_done = true;
+        return std::make_pair(Value(), true);
+    }
+
+    return std::make_pair(value.value(), false);
+}
+
+// https://tc39.es/ecma262/#sec-iterator.prototype.take
+static Value builtinIteratorTake(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    // Let O be the this value.
+    const Value& O = thisValue;
+    // If O is not an Object, throw a TypeError exception.
+    if (!O.isObject()) {
+        ErrorObject::throwBuiltinError(state, ErrorCode::TypeError, "this value is not Object");
+    }
+    // Let iterated be the Iterator Record { [[Iterator]]: O, [[NextMethod]]: undefined, [[Done]]: false }.
+    // Let numLimit be Completion(ToNumber(limit)).
+    // IfAbruptCloseIterator(numLimit, iterated).
+    double numLimit = argv[0].toNumber(state);
+
+    // If numLimit is NaN, then
+    //    a. Let error be ThrowCompletion(a newly created RangeError object).
+    //    b. Return ? IteratorClose(iterated, error).
+    if (std::isnan(numLimit)) {
+        ErrorObject::throwBuiltinError(state, ErrorCode::RangeError, "numLimit is NaN");
+    }
+    // Let integerLimit be ! ToIntegerOrInfinity(numLimit).
+    // If integerLimit < 0, then
+    //    a. Let error be ThrowCompletion(a newly created RangeError object).
+    //    b. Return ? IteratorClose(iterated, error).
+    double integerLimit = toIntegerOrInfinity(numLimit);
```

Comment:
`toIntegerOrInfity` is almost same as `Value::toInteger` function
Is it possible to call `Value::toInteger` instead?